### PR TITLE
KAFKA-9433: Use automated protocol for AlterConfigs request and response

### PR DESCRIPTION
**Pull request by tombentley**

### CODE REVIEW

**Comment by tombentley** [comment]
@mimaison please could you review?

File: clients/src/main/java/org/apache/kafka/common/requests/AlterConfigsRequest.java [hunk 1]
```diff
@@ -104,98 +64,76 @@ public String value() {
...
+        AlterConfigsResponseData data = new AlterConfigsResponseData()
+                .setThrottleTimeMs(throttleTimeMs);
+        for (AlterConfigsRequestData.AlterConfigsResource resource : this.data.resources()) {
+            data.responses().add(new AlterConfigsResponseData.AlterConfigsResourceResponse()
+                    .setResourceType(resource.resourceType())
+                    .setResourceName(resource.resourceName())
+                    .setErrorMessage(error.message())
+                    .setErrorCode(error.error().code()));
+        }
+        return new AlterConfigsResponse(data);
```
Comment (abbccdda) [inline clients/src/main/java/org/apache/kafka/common/requests/AlterConfigsRequest.java:135]: This data construction seems better to be put in the `AlterConfigsResponse` constructor.

**Review by abbccdda — COMMENTED** [review]
Thanks for the PR. Overall LGTM, just one minor comment.

File: clients/src/main/java/org/apache/kafka/common/requests/AlterConfigsRequest.java (hunk 1 above)
Comment (tombentley) [reply clients/src/main/java/org/apache/kafka/common/requests/AlterConfigsRequest.java:135]: Thanks, @abbccdda. There was such a constructor in the `AlterConfigsResponse` originally, but it used a `Map` parameter (and it seemed a bit pointless to me to create a map here just to iterate over it in the constructor). Are you suggesting to revert to that constructor, or have a signiture like `(List<AlterConfigsRequestData.AlterConfigsResource> resources, ApiError error)`?

Comment (abbccdda) [reply clients/src/main/java/org/apache/kafka/common/requests/AlterConfigsRequest.java:135]: Yea, my suggestion would be to reuse the existing constructor as the construction of the `AlterConfigsResponseData` seems non trivial for a caller to do, compared with passing a map of errors.

Comment (tombentley) [reply clients/src/main/java/org/apache/kafka/common/requests/AlterConfigsRequest.java:135]: But it only has this one call site, so the benefit is not clear to me.

**Comment by mimaison** [comment]
ok to test

**Review by mimaison — APPROVED** [review]

**Pull request merged**